packages/react-instantsearch-core-next/src/useIndexContext.ts

```diff
@@ -0,0 +1,15 @@
+import { useContext } from 'react';
+
+import { IndexContext } from './IndexContext';
+
+export function useIndexContext() {
+  const context = useContext(IndexContext);
+
+  if (context === null) {
+    throw new Error(
+      '`useIndexContext` must be used within `IndexContext.Provider`.'
```

Comment:
maybe this error should be "an algolia widget is mounted outside of an InstantSearch provider" (different wording probably) as that's most likely what's wrong? this error seems cryptic for an outsider

Reply:
Right now `useIndexContext` is not exported, but I agree that once it is we need to gather the most frequent misuses to have an appropriate error.

Reply:
ah yes, it's only an index that has this error, not a widget